src/kvstore/NebulaStore.cpp

```diff
@@ -39,7 +39,9 @@ bool NebulaStore::init() {
     LOG(INFO) << "Start the raft service...";
     bgWorkers_ = std::make_shared<thread::GenericThreadPool>();
     bgWorkers_->start(FLAGS_num_workers, "nebula-bgworkers");
-    snapshot_.reset(new SnapshotManagerImpl(this));
+    if (options_.snapManBuilder_) {
```

Comment:
Em....
Any better idea here?

Reply:
Add a `gflag` in SnapshotManagerImpl? That would make this PR very small.

Reply:
I think the key problem is we don't know the key format outside in kvstore.  So the prefix generator should be defined outside.

Reply:
But we still to pass a lambda to NebulaStore, something like:
```
    [] (spaceId, partId) -> std::unique_ptr<KVIterator> iter {
        std::unique_ptr<KVIterator> iter;
        auto ret = store_->prefix(spaceId, partId, prefix, &iter);
        return iter;
    }
```

The KVIterator is what we need here, it seems not a big difference.

Reply:
The difference is for kvstore,  it should not know anything about meta or data.